core/src/components/button/button.tsx

```diff
@@ -159,11 +157,15 @@ export class Button implements ComponentInterface {
   }
 
   hostData() {
-    const { buttonType, disabled, color, expand, hasIconOnly, shape, size, strong } = this;
+    const { buttonType, disabled, color, expand, hasIconOnly, shape, strong } = this;
     let fill = this.fill;
     if (fill === undefined) {
       fill = this.inToolbar ? 'clear' : 'solid';
     }
+    let size = this.size;
```

Comment:
How about `const size = this.size === undefined && this.inItem ? 'small' : this.size`?

Reply:
Sure, updated!